src/coreclr/jit/objectalloc.cpp

```diff
@@ -747,27 +812,30 @@ void ObjectAllocator::ComputeStackObjectPointers(BitVecTraits* bitVecTraits)
                 // Check if this pointer always points to the stack.
                 // For OSR the reference may be pointing at the heap-allocated Tier0 version.
                 //
-                LclVarDsc* lclVarDsc = comp->lvaGetDesc(lclNum);
-
-                if ((lclVarDsc->lvSingleDef == 1) && !comp->opts.IsOSR())
+                if (!comp->opts.IsOSR())
                 {
                     // Check if we know what is assigned to this pointer.
-                    unsigned bitCount = BitVecOps::Count(bitVecTraits, m_ConnGraphAdjacencyMatrix[lclIndex]);
-                    assert(bitCount <= 1);
-                    if (bitCount == 1)
+                    //
+                    bool            isStackPointing = true;
+                    bool            isAssigned      = false;
+                    unsigned        rhsLclIndex     = 0;
+                    BitVecOps::Iter iter(bitVecTraits, m_ConnGraphAdjacencyMatrix[lclIndex]);
+                    while (isStackPointing && iter.NextElem(&rhsLclIndex))
                     {
-                        BitVecOps::Iter iter(bitVecTraits, m_ConnGraphAdjacencyMatrix[lclIndex]);
-                        unsigned        rhsLclIndex = 0;
-                        iter.NextElem(&rhsLclIndex);
-                        unsigned rhsLclNum = IndexToLocal(rhsLclIndex);
-                        if (DoesLclVarPointToStack(rhsLclNum))
-                        {
-                            // The only store to lclNum local is the definitely-stack-pointing
-                            // rhsLclNum local so lclNum local is also definitely-stack-pointing.
-                            MarkLclVarAsDefinitelyStackPointing(lclNum);
-                        }
+                        isAssigned               = true;
+                        const unsigned rhsLclNum = IndexToLocal(rhsLclIndex);
+                        isStackPointing          = DoesLclVarPointToStack(rhsLclNum);
```

Comment:
It seems like we do a lot of work to translate back and forth between "locals" and "indices", which feels a little bit confusing and unnecessary to me. Particularly the fact that "pseudo locals" also are considered "locals" and not just another type of abstract thing that has an index. Do they need to have "local" status?

Reply:
Can the entire loop here be done with `!BitVecOps::IsEmptyIntersection(m_ConnGraphAdjacencyMatrix[lclIndex], m_definitelyStackPointingPointers)`?

Reply:
Yeah, seems like we can avoid looping.

As for lclnum/index shuffling: I could possibl represent the abstract things with only indices and introduce a second set of index-based helpers to avoid some of the back and forth, but I'm not sure that would be an improvement.